docs/DeveloperGuide.md

```diff
@@ -1,37 +1,300 @@
+Command Line Interface (CLI). With the application, users can track their expenses across multiple payment methods,
+calculate monthly expenditure, and set financial goals. The application is targeted at students looking to start
+managing their personal finances.
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Purpose
+This document specifies the architecture and software design decisions for the application, MindMyMoney. 
+The intended audience of this document is the developers, designers, and software testers of MindMyMoney.
+### Acknowledgements
+We would like to thank [AddressBook-3](https://se-education.org/addressbook-level3/) for assisting us in developing
+MindMyMoney.
+### Using the Developer Guide
+Along the guide you may encounter several icons. These icons will provide several useful information.
+> **💡 Note:**
+>- This tells you that there is additional information that is useful when you are using the application.
 
+> **⚠️Warning⚠️**
+> - This tells you that there is some **important** information you should take note of to prevent issues from arising
+    when you are using the application.  
 
-## Product scope
-### Target user profile
+Click on the hyperlinks in the [content page](#Content-Page) to quickly navigate the developer's guide.
 
-{Describe the target user profile}
+## Design
+### Technologies used
+**MindMyMoney** is written fully in **Java 11** using  Object-Oriented Programming (OOP) paradigm to help structure and
+organise the code. This enables the efficiency of future improvements and revisions.
+Data stored in the application is saved into text files locally on the user's device.  
 
-### Value proposition
+### Architecture Overview
+![architecture diagram](images/architecture_diagram.png)  
+Fig 1 - Architecture Diagram for MindMyMoney
 
-{Describe the value proposition: what problem does it solve?}
+The Architecture Diagram above shows the high-level design of the application. The **main components**
+consist of:
+- `MMM`: Initialises the components in the correct sequence and connects them with each other. Also holds the user's
+expenditures in memory.
+- `Ui`: The User Interface of the application and deals with interaction with the user.
+- `Parser`: Deals with making sense of user commands.
+- `Commands`: The collection of all executable commands.
+- `Storage`: Reads data from, and writes data to the hard disk.
 
-## User Stories
+By abstracting out closely related code into classes, it allows `MMM` to deal at a higher level, without worrying
+about the lower level details. Higher cohesion is also achieved and coupling is minimized as each component is
+only coupled to the main class, `MMM`.
+
+The Sequence Diagram below shows an example of how the components interact with each other for the scenario 
+where the user issues the command`add shoes 100`.
+
+![sequence_diagram](images/sequence_diagram.png)
```

Comment:
No critique here, just a really nice diagram sequence that details the architecture of your program!